### PR TITLE
Updated: Pages: Include avatar in Author field.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This PR includes the author avatar on the author field.

Part of: https://github.com/WordPress/gutenberg/issues/63128

cc: @jameskoster 

## Screenshots
Screenshot 2024-07-04 at 18 30 58

## Testing Instructions
Assign an email with a gravatar image to one of the other of a page.
Load the pages list and verify the avatar is correctly shown.

### CODE REVIEW

**Comment by jameskoster** [comment]
Nice, this is working well on Grid and Table layouts. Can we also include it in the List layout? 🤞

**Comment by ellatrix** [comment]
What if I want to turn off the avatars? I've wondered a few times how we could have entries that are composed of multiple fields/values. Another example of this is date and time (I may want to remove the time). Maybe we need a sub-field concept? Or a format concept? For example I could configure a date format, and for authors you could add options for `avatar`, `username`, `first-last`, etc. etc. I also saw @youknowriad [mention](https://github.com/WordPress/gutenberg/pull/63127) a `renderAs` property that is a bit similar.

**Comment by jameskoster** [comment]
That's a valid feature, but potentially something that will get covered in https://github.com/WordPress/gutenberg/issues/58012. I don't know if it needs to hold up this PR?

File: packages/edit-site/src/components/page-templates/author-field.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		<HStack alignment="left" spacing={ 1 }>
+			{ withIcon && imageUrl && (
+				<div
+					className={ clsx( 'page-templates-author-field__avatar', {
+						'is-loaded': isImageLoaded,
+					} ) }
+				>
+					<img
+						onLoad={ () => setIsImageLoaded( true ) }
+						alt=""
```
Comment (carolinan) [inline packages/edit-site/src/components/page-templates/author-field.js]: Why is the alt text empty?

File: packages/edit-site/src/components/posts-app/posts-list.js [hunk 2]
```diff
@@ -395,6 +396,11 @@ export default function PostsList( { postType } ) {
 						value: id,
 						label: name,
 					} ) ) || [],
+				render: ( { item } ) => {
+					return (
```
Comment (ntsekouras) [inline packages/edit-site/src/components/posts-app/posts-list.js:451]: I don't think we should reuse this render field function. Posts list is meant to be reused by every post type eventually, have an API to get the fields by `postType`, but I see templates a bit different than the 'normal' post types. Maybe I'm wrong, but it wouldn't hurt to duplicate just some code for now.

**Review by ntsekouras — COMMENTED** [review]
I think this use case will be handled when we implement the combination of fields, but it's fine to move forwards with this one for now.

File: packages/edit-site/src/components/page-templates/author-field.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/edit-site/src/components/page-templates/author-field.js]: Hi @carolinan good question, it was already empty before in this PR I'm just moving code from the previous AuthorField component https://github.com/WordPress/gutenberg/pull/63142/files#diff-4bc1dbbf462da7ed7ab2b76d40ee01a0a2906482b12bea67383ecbb5697176acL119. But I think it should have an alt text at least something simple as "Author Text" is better than nothing, I made that update.

File: packages/edit-site/src/components/posts-app/posts-list.js (hunk 2 above)
Comment (jorgefilipecosta) [reply packages/edit-site/src/components/posts-app/posts-list.js:451]: Hi @ntsekouras, I applied your suggestion and did not reuse the field from the templates.

**Comment by jorgefilipecosta** [comment]
> I think this use case will be handled when we implement the combination of fields, but it's fine to move forwards with this one for now.

Yes, when we have field combination I can do a small refactor to use that, but I guess for now we can merge this one.

File: packages/edit-site/src/components/posts-app/posts-list.js [hunk 3]
```diff
@@ -207,6 +217,46 @@ function getItemId( item ) {
...
+			const user = getUser( item.author );
+			return {
+				icon: authorIcon,
+				imageUrl: user?.avatar_urls?.[ 48 ],
+				text: user?.name,
+			};
+		},
+		[ item ]
+	);
+	const withIcon = viewType !== LAYOUT_LIST;
```
Comment (ntsekouras) [inline packages/edit-site/src/components/posts-app/posts-list.js]: Nit: let's rename to `withAvatar` for more clarity. Additionally and also nit is to combine the checks like ` {withAvatar && ( imageUrl ? ... : ) }

**Review by ntsekouras — APPROVED** [review]
Left a couple of nits, but this looks good. Thank you!

File: packages/edit-site/src/components/posts-app/posts-list.js [hunk 4]
```diff
@@ -1,7 +1,16 @@
+/**
```
Comment (youknowriad) [inline packages/edit-site/src/components/posts-app/posts-list.js:1]: Unrelated with this PR but this file should be moved to its own components/post-list folder or something, it's not specific to "posts app"

**Comment by jameskoster** [comment]
@jorgefilipecosta it looks like the List layout was missed here. I only see the avatar in the Table or Grid layout.

**Comment by jameskoster** [comment]
@jorgefilipecosta Also, I appreciate this is separate, but in the Templates table the author avatar appears in the Table layout, but not the List layout. It would be good to align across.

**Comment by jorgefilipecosta** [comment]
Hi @jameskoster with the iterations we did there are avatars, on list, grid, and table on templates and pages. So everything you pointed out should be fixed. Let me know if there is anything related remaining.

**Pull request merged**